--- a/libopkg/opkg_download.c
+++ b/libopkg/opkg_download.c
@@ -157,11 +157,15 @@ opkg_download(const char *src, const cha
 #else
     {
       int res;
-      const char *argv[8];
+      const char *argv[11];
       int i = 0;
 
       argv[i++] = "wget";
       argv[i++] = "-q";
+      argv[i++] = "--private-key=/etc/ssl/certs/egwclient.key";
+      argv[i++] = "--certificate=/etc/ssl/certs/egwclient.crt";
+      argv[i++] = "--no-check-certificate"; 
+                  
       if (conf->http_proxy || conf->ftp_proxy) {
 	argv[i++] = "-Y";
 	argv[i++] = "on";
